src/app/core/services/issue.service.ts

```diff
@@ -147,6 +147,11 @@ export class IssueService {
         }),
         catchError((err) => {
           this.logger.error(err); // Log full details of error first
+          if (err.status === 422) {
+            // Custom error message for failed validation
+            return throwError(`Unable to assign the selected user(s) to this issue.
+              Possible reason: a selected user has not joined the GitHub organization ${err.request.data.owner} yet`);
```

Comment:
This is giving me a different error: `TypeError: Cannot read properties of undefined (reading 'owner')`

This means that err.request.data is null. You can get the organization name from phase service or github service.

Also, are all http 422 errors guaranteed to be due to wrong assignees? Maybe we should parse the error message from github to check if it is really due to assignees?
Example of the error: `Validation Failed: {"value":"fakestudent","resource":"Issue","field":"assignees","code":"invalid"}`

Reply:
hi @chunweii , thanks for the review, may I check that `Validation Failed: {"value":"fakestudent","resource":"Issue","field":"assignees","code":"invalid"}` is a string, and if possible, could I have the error schema? Thank you!

Reply:
Hi @cedricongjh, sorry for the late reply. That message is a string from github. You can also get the json object from the error. You can try adding `fakestudent` as an assignee to see the error logged in the console. Check https://docs.github.com/en/rest/overview/resources-in-the-rest-api#client-errors for more info